price-estimator/src/estimate_buy_amount.rs

```diff
@@ -0,0 +1,20 @@
+use crate::filter::TokenPair;
+use pricegraph::Orderbook;
+
+pub fn estimate_buy_amount(
+    token_pair: TokenPair,
+    sell_amount_in_quote: f64,
+    price_rounding_buffer: f64,
+    orderbook: Orderbook,
```

Comment:
Is there a reason for this very function oriented approach? I would have expected a BuyAmountEstimator struct that depends on the orderbook and price_rounding_buffer and exposes `estimate_buy_amount(token_pair, sell_amount_in_quote)`.

Not that it makes a huge difference, but I'd assume for more complex handlers it can become more messy to pass all per Instance state around.

Reply:
I do prefer this to be a function from the perspective of users of the module (the one public fn) but there is no reason that it couldn't be a struct internally. Whether we bundle this as a struct or pass parameters makes no functional difference, it just makes the signatures cleaner. If there were more functions or more parameters I'd probably do it.

Reply:
I slightly prefer a function here as well for the handler, however one suggestion would be to group up some of the global state into some `ServiceContext` struct with the current `Orderbook` and the service options like `price_rounding_buffer`. On the other hand, I also agree with @e00E  that we should add a couple more handlers first, so we have a better idea of what patterns emerge.